src/main/java/com/cloudbees/jenkins/support/api/FileContent.java

```diff
@@ -25,27 +25,37 @@
 package com.cloudbees.jenkins.support.api;
 
 import com.cloudbees.jenkins.support.SupportLogFormatter;
+import com.cloudbees.jenkins.support.filter.ContentFilter;
+import com.cloudbees.jenkins.support.filter.PrefilteredContent;
+import com.cloudbees.jenkins.support.util.StreamUtils;
 import org.apache.commons.io.IOUtils;
 
+import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.FilterInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.io.OutputStreamWriter;
 import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
 
 /**
  * Content that is stored as a file on disk.
  *
  * @author Stephen Connolly
  */
-public class FileContent extends Content {
+public class FileContent extends PrefilteredContent {
 
     protected final File file;
     private final long maxSize;
+    private final boolean isBinary;
+
+    private final static String ENCODING = "UTF-8";
```

Comment:
I assume having this upper or lower case is the same right?

Reply:
Yes. Also `Charset` class waits for "UTF-8"